main/src/cgeo/geocaching/apps/navi/OsmAndApp.java

```diff
@@ -0,0 +1,97 @@
+package cgeo.geocaching.apps.navi;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.activity.ActivityMixin;
+import cgeo.geocaching.apps.AbstractApp;
+import cgeo.geocaching.location.Geopoint;
+import cgeo.geocaching.models.Geocache;
+import cgeo.geocaching.models.Waypoint;
+import cgeo.geocaching.utils.ProcessUtils;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class OsmAndApp extends AbstractApp implements CacheNavigationApp, WaypointNavigationApp, GeopointNavigationApp {
+
+    private static final String PARAM_NAME = "name";
+    private static final String PARAM_LAT = "lat";
+    private static final String PARAM_LON = "lon";
+    private static final String PREFIX = "osmand.api://";
+    private static final String GET_INFO = "get_info";
+    private static final String ADD_MAP_MARKER = "add_map_marker";
+
+    protected OsmAndApp() {
+        super(getString(R.string.cache_menu_osmand), null);
+    }
+
+    @Override
+    public boolean isInstalled() {
+        return ProcessUtils.isIntentAvailable(Intent.ACTION_VIEW, Uri.parse(PREFIX + GET_INFO));
+    }
+
+    @Override
+    public boolean isEnabled(@NonNull final Geocache cache) {
+        return cache.getCoords() != null;
+    }
+
+    @Override
+    public boolean isEnabled(@NonNull final Waypoint waypoint) {
+        return waypoint.getCoords() != null;
+    }
+
+    @Override
+    public void navigate(@NonNull final Activity activity, @NonNull final Geocache cache) {
+        if (cache.getCoords() != null) {

```

Comment:
I suggest to derive from AbstractPointNavigationApp, to avoid duplicating the "isEnabled" methods.

@samueltardieu  The coords cannot be null here due to constraints outside this method, however static code analysis dictates that they may be null. Therefore the if-else is acceptable. We may want to refactor all Apps after merging to just use an assertion inside the method instead of if-else to silence static code analysis.